libraries/core_libs/consensus/src/transaction/transaction_queue.cpp

```diff
@@ -52,18 +73,50 @@ bool TransactionQueue::erase(const trx_hash_t &hash) {
   return true;
 }
 
-bool TransactionQueue::insert(std::shared_ptr<Transaction> &&transaction) {
-  assert(transaction);
-  if (hash_queue_.contains(transaction->getHash())) return false;
+bool TransactionQueue::insert(std::pair<std::shared_ptr<Transaction>, TransactionStatus> &&transaction,
+                              uint64_t last_block_number) {
+  assert(transaction.first);
+  if (hash_queue_.contains(transaction.first->getHash())) return false;
```

Comment:
Is this check supposed to protect tx queue before duplicate insertion ? If yes, it wont work properly as hash_queue_ is adjusted only in case of TransactionStatus::Verified, not for LowNonce and InsufficentBalance. Also it is somehow related to some comparator but that seems ok that we add txs there only tx status == Verified

@MatusKysel please confirm

Reply:
I changed it to check for both lownonce and insufficientbalance.

Reply:
it is fine now